nc-times/src/components/AllArticles.jsx

```diff
@@ -3,19 +3,24 @@ import { getAllArticles } from "../api"
 import Loading from "../resources/Loading"
 import ArticleCard from "../resources/ArticleCard"
 import TopicsBar from './TopicsBar'
+import ErrorPage from "./ErrorPage"
 
 function AllArticles() {
     const [articles, setArticles] = useState([]);
     const [loading, setLoading] = useState(true);
+    const [error, setError]= useState(null)
 
     useEffect(() => {
         getAllArticles()
             .then((data) => {
                 setLoading(false);
                 setArticles(data);
             })
+            .catch((err)=>{setError({err})})
     }, []);
-
+if(error){ 
+    return (<ErrorPage/>)
```

Comment:
You could make the error page more dynamic by passing in props that tell the user a bit more about what's gone wrong. For example a message that could then be slotted into your error component.